git-pr-merge

```diff
@@ -65,40 +87,95 @@ prmerge::setup() {
   fi
 }
 
-prmerge::prepare() {
-  echo merging "${feature}" to "${master}"
-
-  # Fetch from all remotes so we can check if local branch is up-to-date
-  git remote update >/dev/null
-
-  if [[ "$(git rev-parse "${master}")" != "$(git rev-parse "${master}@{upstream}")" ]]; then
-    printf '%s is not up-to-date. You should run:\n' "${master}" >&2
-    printf '  git checkout %s\n' "${master}" >&2
-    printf '  git pull\n' >&2
-    return 1
-  fi
-
+common::prepare() {
   # cd to root of repo as master may not contain CWD, but switch back
   # when we're done.
   # shellcheck disable=SC2064
   # We want to evaluate $(pwd) now and not later.
   trap "cd \"$(pwd)\"" EXIT
   cd "$(git rev-parse --show-toplevel)"
 
+  # Fetch from all remotes so we can check if local branch is up-to-date
+  git remote update >/dev/null
+}
+
+prmerge::prepare() {
+  common::prepare
+
+  local do_pull=false
+  if [[ "$(git rev-parse "${master}")" != "$(git rev-parse "${master}@{upstream}")" ]]; then
+    printf '\n%s is not up-to-date. Press enter to update:' "${master}"
+    read -r
+    do_pull=true
+  fi
+
   git checkout -q "${master}"
+  "${do_pull}" && git pull
+}
+
+prupdate::prepare() {
+  common::prepare
+
+  # If the head of ${feature} is a merge commit from previously running
+  # git-pr-update, reset that commit and force push before generating the
+  # next merge commit. This seems to stop github removing approval on a PR.
+  read -ra feature_parents < <(git show --no-patch --format='%P' "${feature}" --)
+  if (( ${#feature_parents[@]} == 2 )); then
```

Comment:
I dont think so. If you have an octopus merge at the head of your branch, this should not stop you doing the update. Ideally this check would be more accurate and properly identify when there is a merge commit at HEAD that was the result of a git-pr-update - they are the only ones we want to undo, but I need to think about that more. It may be as simple as `parent[0]` being an ancestor of `${master}` but I want to think about that a little more.

Reply:
did the ancestor checking